sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,1142 @@
+               SWITCHING_MODE_AUTO if auto switch is enabled.
+               SWITCHING_MODE_MANUAL if manual switch is enabled.
+        """
+
+        raise NotImplementedError
+
+    def get_nic_temperature(self):
+        """
+        This API returns nic temperature of the physical port for which this API is called.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            an Integer, the temperature of the NIC MCU
+        """
+
+        raise NotImplementedError
+
+    def get_local_temperature(self):
+        """
+        This API returns local ToR temperature of the physical port for which this API is called.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            an Integer, the temperature of the local MCU
+        """
+
+        raise NotImplementedError
+
+    def get_nic_voltage(self):
+        """
+        This API returns nic voltage of the physical port for which this API is called.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            a float, the voltage of the NIC MCU
+        """
+
+        raise NotImplementedError
+
+    def get_local_voltage(self):
+        """
+        This API returns local ToR voltage of the physical port for which this API is called.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            a float, the voltage of the local MCU
+        """
+
+        raise NotImplementedError
+
+    def get_alive_status(self):
+        """
+        This API checks if cable is connected to all the ports and is healthy.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            a boolean, True if the cable is alive
+                     , False if the cable is not alive
+        """
+
+        raise NotImplementedError
+
+    def reset(self, target):
+        """
+        This API resets the MCU to which this API is called for.
+        The target specifies which MCU is supposed to be reset
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            target:
+                One of the following predefined constants, the actual target to check the link on:
+                     TARGET_NIC -> NIC,
+                     TARGET_TOR_A -> TORA,
+                     TARGET_TOR_B -> TORB
+
+        Returns:
+            a boolean, True if the cable is target reset
+                     , False if the cable target is not reset
+        """
+
+        raise NotImplementedError
+
+    def create_port(self, speed, fec_mode_tor_a=FEC_MODE_NONE, fec_mode_tor_b=FEC_MODE_NONE,
+                    fec_mode_nic=FEC_MODE_NONE, anlt_tor_a=False, anlt_tor_b= False, anlt_nic=False):
```

Comment:
Is it make sense to change the input parameter as suggestion?
For current y_cable firmware implementation, both TOR A and TOR B must be operated in the same mode. it is not allowed to configure AN/LT, FEC on TOR A and B individually.

```suggestion
                    fec_mode_nic=FEC_MODE_NONE, anlt_tor=False, anlt_nic=False):
```

Reply:
fixed, can you please check